### PR TITLE
Factor some common `io::Error` constants

### PR DESCRIPTION
**Pull request by a1phyr**

### CODE REVIEW

**Comment by cuviper** [comment]
That's a nice cleanup, thanks!

@bors r+

**Comment by tgross35** [comment]
This looks to have failed in the rollup, https://github.com/rust-lang-ci/rust/actions/runs/8637707189/job/23680527190#step:27:3291

```
   error: associated constant `ZERO_TIMEOUT` is never used
    --> library/std/src/io/error.rs:93:22
     |
  79 | impl Error {
     | ---------- associated constant in this implementation
  ...
  93 |     pub(crate) const ZERO_TIMEOUT: Self =
     |                      ^^^^^^^^^^^^
     |
     = note: `-D dead-code` implied by `-D warnings`
     = help: to override `-D warnings` add `#[allow(dead_code)]`
  
  [RUSTC-TIMING] std test:false 2.337
  warning: `std` (lib) generated 1 warning
  error: could not compile `std` (lib) due to 1 previous error; 1 warning emitted
```

File: library/std/src/io/error.rs [hunk 1]
```diff
@@ -75,6 +75,25 @@ impl fmt::Debug for Error {
...
 
+/// Common errors constants for use in std
+impl Error {
+    pub(crate) const INVALID_UTF8: Self =
+        const_io_error!(ErrorKind::InvalidData, "stream did not contain valid UTF-8");
+
+    pub(crate) const READ_EXACT_EOF: Self =
+        const_io_error!(ErrorKind::UnexpectedEof, "failed to fill whole buffer");
+
+    #[allow(dead_code)]
```
Comment (cuviper) [inline library/std/src/io/error.rs:86]: I think it would be fine to lift this onto the entire `impl` block.

**Pull request merged**